From 947505850bdea3d1feb4e3e846e5433970bd2022 Mon Sep 17 00:00:00 2001
From: Peng Sun <peng.p.sun@intel.com>
Date: Fri, 3 Sep 2021 21:20:40 +0800
Subject: [PATCH] Masquerade to KVM hypervisor type.

Signed-off-by: Peng Sun <peng.p.sun@intel.com>
---
 hypervisor/arch/x86/guest/vcpuid.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/hypervisor/arch/x86/guest/vcpuid.c b/hypervisor/arch/x86/guest/vcpuid.c
index bc19d0481..db0d669cb 100644
--- a/hypervisor/arch/x86/guest/vcpuid.c
+++ b/hypervisor/arch/x86/guest/vcpuid.c
@@ -291,8 +291,15 @@ static int32_t set_vcpuid_extended_function(struct acrn_vm *vm)
 	uint32_t i, limit;
 	struct vcpuid_entry entry;
 	int32_t result;
+	static const char sig[12] = "KVMKVMKVM";
+	const uint32_t *sigptr = (const uint32_t *)sig;
 
 	init_vcpuid_entry(0x40000000U, 0U, 0U, &entry);
+	if (!is_sos_vm(vm)) {
+		entry.ebx = sigptr[0];
+		entry.ecx = sigptr[1];
+		entry.edx = sigptr[2];
+	}
 	result = set_vcpuid_entry(vm, &entry);
 	if (result == 0) {
 		init_vcpuid_entry(0x40000001U, 0U, 0U, &entry);
-- 
2.17.1

